thunder/torch/__init__.py

```diff
@@ -186,6 +186,92 @@ def is_cuda(a: TensorLike, /) -> bool:
 
 register_method("size", size)
 
+_torch_dtype_to_old_torch_typestring_map = {
+    torch.float32: "FloatTensor",
+    torch.float64: "DoubleTensor",
+    torch.float16: "HalfTensor",
+    torch.bfloat16: "BFloat16Tensor",
+    torch.uint8: "ByteTensor",
+    torch.int8: "CharTensor",
+    torch.int16: "ShortTensor",
+    torch.int32: "IntTensor",
+    torch.long: "LongTensor",
+    torch.bool: "BoolTensor",
+}
+
+_old_torch_typestring_to_torch_dtype_map = {v: k for k, v in _torch_dtype_to_old_torch_typestring_map.items()}
+
+
+def _device_and_dtype_to_old_torch_typestring(device: DeviceLike, dtype: dtypeLike):
+    torch_dtype = to_torch_dtype(dtype)
+    dtype_str = _torch_dtype_to_old_torch_typestring_map.get(torch_dtype)
+    devicetype_str: str = ""
+    if device.devicetype is not devices.DeviceType.CPU:
+        devicetype_str = f"{devices.devicetype_string(device.devicetype)}."
+    return f"torch.{devicetype_str}{dtype_str}"
+
+
+def _old_torch_typestring_to_devicetype_and_dtype(typestring: str) -> tuple[DeviceLike, dtypeLike]:
+
+    # Two cases:
+    #    - torch.DtypeTensor
+    #    - torch.device.DtypeTensor
+
+    _, *dev_and_dtype = typestring.split(".")
+
+    if len(dev_and_dtype) == 1:
+        (dtype_str,) = dev_and_dtype
+        return "cpu", _old_torch_typestring_to_torch_dtype_map[dtype_str]
+
+    if len(dev_and_dtype) == 2:
+        dtype_str, devicetype_str = dev_and_dtype
```

Comment:
Aren't these two swapped? Won't `dtype_str` be 'cuda'?

Reply:
If these two are swapped, then how are the OpInfo tests passing?

Reply:
I don't think these two are swapped. For instance, when the user inputs `torch.IntTensor` to a CUDA tensor, this will result in a tensor on a CPU device. So, if `len(dev_and_dtype) == 1`, I think it is safe to assume it is going to copy the tensor on a CPU device.

Reply:
I understand what you're saying, but I think what I said was unclear, sorry. Let me rephrase my earlier comment. The original code is:

```
# Two cases:
    #    - torch.DtypeTensor
    #    - torch.device.DtypeTensor

    _, *dev_and_dtype = typestring.split(".")
```

at this point, won't `dev_and_dtype` be `(device, DtypeTensor)` or `(DtypeTensor,)`?

```
    if len(dev_and_dtype) == 1:
        (dtype_str,) = dev_and_dtype
        return "cpu", _old_torch_typestring_to_torch_dtype_map[dtype_str]
```

this block handles the case that `dev_and_dtype` is `(DtypeTensor,)`, and then the next block

```
    if len(dev_and_dtype) == 2:
        dtype_str, devicetype_str = dev_and_dtype
```

wouldn't `dev_and_dtype` be `(device, DtypeTensor)` here, but it's unpacked as `(dtype_str, devicetype_str)`, and that's what I'm confused by. I would expect the unpacking to be `devicetype_str, dtype_str`?

Reply:
Oh my bad, I completely misunderstood your point. Yes, you are right! let me update this. They were swapped indeed 😅

Reply:
And to answer your above question, as I was doing something like this:
```
_old_torch_typestring_to_torch_dtype_map[devicetype_str]
```
it was passing the test! My apologies for the confusion.